### PR TITLE
geo/wkt: add support for parsing linestrings with Z and M dimensions

### PR DESCRIPTION
**Pull request by andyyang890**
This patch extends the capabilities of the WKT parser to include
parsing of linestrings with Z and M dimensions. This patch also
improves the error messages generated for parsing errors.

Refs: #53091

Release note: None

### CODE REVIEW

File: pkg/geo/wkt/wkt.y [hunk 1]
```diff
@@ -42,31 +44,38 @@ start:
 
 geometry:
 	point
+|	linestring
+	{
+		// Check that non-empty linestrings have at least 2 points.
```
Comment (andyyang890) [inline pkg/geo/wkt/wkt.y:49]: @otan Should I take this out?

Comment (otan) [reply pkg/geo/wkt/wkt.y:49]: yeah i think so, especially if we're not returning a more descriptive error message here.

File: pkg/geo/wkt/wkt.y [hunk 2]
```diff
@@ -24,13 +24,15 @@ import "github.com/twpayne/go-geom"
 }
 
 %token <str> POINT POINTZ POINTM POINTZM
+%token <str> LINESTRING LINESTRINGZ LINESTRINGM LINESTRINGZM
```
Comment (otan) [inline pkg/geo/wkt/wkt.y:27]: super nit: define these in alphabetical order

File: pkg/geo/wkt/wkt_test.go [hunk 3]
```diff
@@ -98,11 +141,22 @@ func TestUnmarshalError(t *testing.T) {
...
 		},
+		{
+			desc:           "2D linestring with only one point",
+			input:          "LINESTRING(0 0)",
+			expectedErrStr: `parse error: could not parse "LINESTRING(0 0)"`,
+		},
+		{
+			desc:           "linestring with mixed dimensionality",
+			input:          "LINESTRING(0 0, 1 1 1)",
+			expectedErrStr: `parse error: could not parse "LINESTRING(0 0, 1 1 1)"`,
```
Comment (otan) [inline pkg/geo/wkt/wkt_test.go:158]: can we return the position in the error message here?

the unfortunate side effect of doing it the way we're doing it now is we cannot give a great error message when there's mixed dimensionality but should be able to somehow tell this to the user. maybe worth a TODO.

File: pkg/geo/wkt/generate.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+goyacc -o wkt_generated.go -p "wkt" wkt.y
+#todo (ayang) modify goyacc to accept a cmd line flag to set this
```
Comment (otan) [inline pkg/geo/wkt/generate.sh:4]: nit: format TODO as `# TODO(ayang):`

**Review by otan — APPROVED** [review]
this is really awesome!

File: pkg/geo/wkt/generate.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+sed -i '' 's/wktErrorVerbose = false/wktErrorVerbose = true/' wkt_generated.go
```
Comment (otan) [inline pkg/geo/wkt/generate.sh:5]: you'll need to make this `sed` linux compatible i think (`sed -i ''` on OSX/BSD, `sed -i` on everything else). may be easier to use `awk`... or write a go wrapper that does the replace and thus environment proof, e.g.

generate_wkt.go
```
// +build do_not_build

package main

func main() {
   // run goyacc
   // replace regex and write file
}
```

and then
```
//go:generate go run generate_wkt.go
```

Comment (andyyang890) [reply pkg/geo/wkt/generate.sh]: Discussed offline that using `sed` with a temp file should work

**Review by otan — APPROVED** [review]

**Comment by andyyang890** [comment]
bors r=otan

**Pull request merged**